tests/helper/helper_http.go

```diff
@@ -47,3 +50,27 @@ func HttpWaitForWithStatus(url string, match string, maxRetry int, interval int,
 	fmt.Fprintf(GinkgoWriter, "Last output from %s: %s\n", url, string(body))
 	Fail(fmt.Sprintf("Failed after %d retries. Content in %s doesn't include '%s'.", maxRetry, url, match))
 }
+
+// GetRandomFreePort returns a random free port(string) between 1024 and 65535, or within a given portRange if provided
+// WARN: If length of portRange is anything other than 2 and first index is greater-than-or-equal-to second index, it will use the default range.
+func GetRandomFreePort(portRange ...int) string {
+	// rand.Seed() has been deprecated from Go 1.20, and it has been advised to use the following method instead
+	randomizer := rand.New(rand.NewSource(int64(time.Now().Nanosecond()))) // #nosec  // cannot use crypto/rand library here
```

Comment:
Adding a seed is useful to not have the same series of number every time. But in this case, we don't care about the reproducibility of this series, we just want to have different ports between subsequent calls to rand.Intn. So, I would try bby using rand.Intn without any previous call to Seed or equivalent.

Another thing, Seed is normally called only once, to start a new sequence of pseudo-random numbers. Here, you are re-initializing it every time you pick a new number, so you are getting every time the first number of the series.

Reply:
To illustrate what I mean, it could be better to use a global counter starting from 30000 across all the tests and it should work. The series would always be 30000, 30001, 30002, ..., but you are sure you never get the same port during the series.

Reply:
> To illustrate what I mean, it could be better to use a global counter starting from 30000 across all the tests and it should work. The series would always be 30000, 30001, 30002, ..., but you are sure you never get the same port during the series.

Do you mean I should simply try to use the way `odo dev`? Start from a counter(30000), set a range(40000), and check for the next free port?

If that is true, then it won't work in case we're calling the function back to back without calling `odo dev`, it will return the same port number because it is still free, and will be until `odo dev` starts. This was the reason I wanted to use random port numbers.

>Adding a seed is useful to not have the same series of number every time. But in this case, we don't care about the reproducibility of this series, we just want to have different ports between subsequent calls to rand.Intn. So, I would try bby using rand.Intn without any previous call to Seed or equivalent.

You're right, we can do without the Seed.

Reply:
> Do you mean I should simply try to use the way odo dev? Start from a counter(30000), set a range(40000), and check for the next free port?

No, I mean a global value that would increment itself each time you get its value, in an atomic way.

Reply:
> > Do you mean I should simply try to use the way odo dev? Start from a counter(30000), set a range(40000), and check for the next free port?
> 
> No, I mean a global value that would increment itself each time you get its value, in an atomic way.

Okay, I'll try that.

Reply:
This was to illustrate my meaning about seed. If it works by removing the seed, I think it's ok